epsagon/events/sqlalchemy.py

```diff
@@ -0,0 +1,89 @@
+"""
+sqlalchemy events module.
+"""
+
+from __future__ import absolute_import
+import traceback
+from uuid import uuid4
+
+from ..trace import trace_factory
+from ..event import BaseEvent
+from ..utils import database_connection_type
+
+
+class SqlAlchemyEvent(BaseEvent):
+    """
+    Represents base SqlAlchemy event.
+    """
+
+    ORIGIN = 'sqlalchemy'
+    RESOURCE_TYPE = 'database'
+
+    # pylint: disable=W0613
+    def __init__(self, wrapped, instance, args, kwargs, start_time, response,
+                 exception, operation):
+        """
+        Initialize.
+        :param wrapped: wrapt's wrapped
+        :param instance: wrapt's instance
+        :param args: wrapt's args
+        :param kwargs: wrapt's kwargs
+        :param start_time: Start timestamp (epoch)
+        :param response: response data
+        :param exception: Exception (if happened)
+        :param operation: sqlalchemy operation
+        """
+        super(SqlAlchemyEvent, self).__init__(start_time)
+
+        self.event_id = 'sqlalchemy-{}'.format(str(uuid4()))
+
+        self.resource['name'] = (
+                instance.bind.url.database or instance.bind.url.host
+        )
+        self.resource['operation'] = operation
+
+        # override event type with the specific DB type
+        self.resource['type'] = database_connection_type(
+            instance.bind.url.host,
+            self.RESOURCE_TYPE
+        )
+
+        if exception is not None:
+            self.set_exception(exception, traceback.format_exc())
+
```

Comment:
don't we wanna save the response? (if metadata only is false)

Reply:
There's nothing interesting in the response here

Reply:
right, we just catch init & close sessions